chatcommands.py

```diff
@@ -1719,219 +1717,148 @@ def allspam(msg, url):
-                    if GlobalVars.api_backoff_time < time.time() + answer_res["backoff"]:
-                        GlobalVars.api_backoff_time = time.time() + answer_res["backoff"]
-                GlobalVars.api_request_lock.release()
-                # Finally, set the attribute
-                post_data.question_id = answer_res['items'][0]['question_id']
-                post_data.is_answer = True
-            user_posts.append(post_data)
-    if len(user_posts) == 0:
-        raise CmdException("The specified user hasn't posted anything.")
-    if len(user_posts) > 15:
+            user_post_urls.append(post['link'])
+    if len(user_post_urls) == 0:
+        raise CmdException("The user has no post yet.")
+    if len(user_post_urls) > 15:
         raise CmdException("The specified user has an abnormally high number of spam posts. Please consider flagging "
                            "for moderator attention, otherwise use !!/report on the posts individually.")
-    why_info = u"User manually reported by *{}* in room *{}*.\n".format(msg.owner.name, msg.room.name)
-    # Handle all posts
-    for index, post in enumerate(user_posts, start=1):
-        batch = ""
-        if len(user_posts) > 1:
-            batch = " (batch report: post {} out of {})".format(index, len(user_posts))
-        handle_spam(post=Post(api_response=post.as_dict),
-                    reasons=["Manually reported " + post.post_type + batch],
-                    why=why_info)
-        time.sleep(2)  # Should this be implemented differently?
-    if len(user_posts) > 2:
+
+    output = []
+    for current_url in user_post_urls:
+        # Report that post
+        current_output = report_post(current_url, msg.owner, msg.room.name, message_url, operation, custom_reason)
+        if current_output:
+            output.append(current_output)
+    if len(user_post_urls) > 2:
         add_or_update_multiple_reporter(msg.owner.id, msg._client.host, time.time())
+    if len(output):
+        return "\n".join(output)
 
 
-def report_posts(urls, reported_by_owner, reported_in=None, blacklist_by=None, operation="report", custom_reason=None):
-    """
-    Reports a list of URLs
-    :param urls: A list of URLs
-    :param reported_by_owner: The chatexchange User record for the user that reported the URLs.
-    :param reported_in: The name of the room in which the URLs were reported, or True if reported by the MS API.
-    :param blacklist_by: String of the URL for the transcript of the chat message causing the report.
-    :param operation: String of which operation is being performed (e.g. report, scan, report-force, scan-force)
-    :param custom_reason: String of the custom reason why the URLs are being reported.
-    :return: String: the in-chat repsponse
-    """
-    reported_by_name = reported_by_owner.name
+def report_post(url, reported_by, reported_in=None, blacklist_by=None, operation="report", custom_reason=None):
+    """ Scan or report a single post """
```

Comment:
The portion of the docstring which described the parameters and return value had value here. While I don't feel there's a requirement to have such on every function, if it already exists, at least mostly, then it's reasonable for us to keep them (with changes to what the function now uses).

Reply:
After all, the fact that such information wasn't documented well was a contributing factor to the bug where a report reason didn't end up as an MS comment (i.e. the wrong thing was being passed, in part because it wasn't clear what was needed).

Reply:
Supposedly docstring is machine oriented, as it may be accessed by `.__doc__`. Therefore I tries to make docstring consistent (but did not really done a good job). However I added comments there for documentation. This should also go to refdoc.md